geomstats/geometry/heisenberg.py

```diff
@@ -0,0 +1,204 @@
+"""Implementation of the 3D Heisenberg group."""
+
+import geomstats.backend as gs
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.lie_group import LieGroup
+
+
+class heisenbergVectors(LieGroup):
+    r"""Class for the 3D Heisenberg group in vector representation.
+
+    The 3D Heisenberg group represented as R^3. It is a step-2 Carnot Lie
+    group. It can be equipped with a natural sub-Riemannian structure, and it is
+    it a fundamental example in sub-Riemannian geometry.
+
+    Parameters
+    ----------
+    No parameters
+
+    Reference
+    ---------
+    https://en.wikipedia.org/wiki/Heisenberg_group
+
+    """
+
+    def __init__(self, **kwargs):
+        super(heisenbergVectors, self).__init__(
+            dim=int(3), default_point_type='vector',
+            lie_algebra=Euclidean(3))
+
+    def get_identity(self, point_type='vector'):
+        """Get the identity of the 3D Heisenberg group.
+
+        Parameters
+        ----------
+        point_type : str, {'vector', 'matrix'}
+            Point_type of the returned value. Unused here.
+
+        Returns
+        -------
+        get_identity : array-like, shape=[1,]
+        """
+        return gs.zeros(self.dim)
+
+    identity = property(get_identity)
+
+    def belongs(self, point):
+        """Evaluate if a point belongs to the 3D Heisenberg group.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point to evaluate.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the Heisenberg
+            group (in this representation this is just R^3).
+        """
+        point_dim = point.shape[-1]
+        belongs = point_dim == self.dim
+        if gs.ndim(point) == 2:
+            belongs = gs.tile([belongs], (point.shape[0],))
+
+        return belongs
+
+    def compose(self, point_a, point_b):
+        """Compute the group product of elements `point_a` and `point_b`.
+
+        Parameters
+        ----------
+        point_a : array-like, shape=[..., {dim, [n, n]}]
+            Left factor in the product.
+        point_b : array-like, shape=[..., {dim, [n, n]}]
+            Right factor in the product.
+
+        Returns
+        -------
+        composed : array-like, shape=[..., {dim, [n, n]}]
+            Product of point_a and point_b along the first dimension.
+        """
+        return (point_a + point_b +
+                1 / 2 * gs.array([0, 0, point_a[0] *
+                                  point_b[1] - point_a[1] *
+                                  point_b[0]]))
```

Comment:
this does not look vectorized? Consider that point_a could be a list of inputs -> use point_a[..., 0]
Can this term be computed using det or cross product?